### PR TITLE
Don't put an underscore in the version portion of Python 3.10 compatibility tags

### PR DESCRIPTION
**Pull request by hroncok**
PEP 641 is rejected.

See https://discuss.python.org/t/5513/42
See https://www.python.org/dev/peps/pep-0641/

Fixes https://github.com/pypa/packaging/issues/372

### CODE REVIEW

**Comment by hroncok** [comment]
Unsurprisingly, when this is patched in wheel (vendors packaging.tags) and pip (vendors packaging), it fixes the current breakage (tested couple Fedora packages that were affected). Until (and unless) PEP 641 is approved and implemented, maybe this is a way to go to unblock testing with Python 3.10 early alphas?

**Comment by brettcannon** [comment]
I would rather not have to ask the packaging ecosystem to do multiple updates just for this if possible. It doesn't matter _quite_ yet anyway, since any new release is held up on macOS 11/Arm support anyway.

**Comment by mattip** [comment]
fixes gh-372. Since the PEP was rejected, I think this can move forward.

File: packaging/tags.py
Comment (pradyunsg) [inline packaging/tags.py:799]: nit:

```suggestion
    return "".join(map(str, version))
```

**Comment by hroncok** [comment]
Rebased and updated the commit message. Included suggestion by @pradyunsg (thanks).

**Review by di — APPROVED** [review]

**Review by brettcannon — APPROVED** [review]

**Pull request merged**
